crates/uniffi-bindgen-react-native/src/bindings/react_native/gen_typescript/templates/ObjectTemplate.ts

```diff
@@ -84,11 +83,35 @@ export class {{ impl_class_name }} implements {{ protocol_name }}, UniffiObjectI
 
 }
 
-function create{{ impl_class_name }}(pointer: UnsafeMutableRawPointer): {{ impl_class_name }} {
-    const instance = Object.create({{ impl_class_name }}.prototype);
-    instance.pointer = pointer;
-    return instance;
-}
+const {{ obj_factory }}: UniffiObjectFactory<{{type_name}}> = {
+    create(pointer: UnsafeMutableRawPointer): {{ type_name }} {
+        const instance = Object.create({{ type_name }}.prototype);
+        instance._rustArcPtr = this.bless(pointer);
+        return instance;
+    },
+
+    bless(pointer: UnsafeMutableRawPointer): UniffiRustArcPtr {
+        const destructor = this.freePointer;
+        return uniffiBlessPointer(pointer, destructor);
+    },
+
+    pointer(obj: {{ type_name }}): UnsafeMutableRawPointer {
+        const ptr = (obj as any)._rustArcPtr;
+        if (ptr === undefined) {
+            throw new UniffiInternalError.UnexpectedNullPointer();
+        }
+        return ptr.p;
+    },
+
+    clonePointer(obj: {{ type_name }}): UnsafeMutableRawPointer {
+        const pointer = this.pointer(obj);
+        return rustCall(callStatus => NativeModule.{{ obj.ffi_object_clone().name() }}(pointer, callStatus) );
+    },
+
+    freePointer(pointer: UnsafeMutableRawPointer): void {
```

Comment:
Actually, looking at freePointer it doesn't seem to actually use any `this` methods, so it might actually not need the bind after all?  Though I don't know if javascript will have trouble with the missing hidden-first-argument for `this` that it expects.

Reply:
You're right: the destructor doesn't need the `this`. In this case, `this` when the `freePointer` function is called from C++ is unset. AFAIK, javascript has it as the `this` value, rather than a hidden-first-argument.